phoenix-core/src/main/java/org/apache/phoenix/schema/stats/UpdateStatisticsTool.java

```diff
@@ -0,0 +1,212 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.schema.stats;
+
+import org.antlr.runtime.CharStream;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.PosixParser;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.mapreduce.TableMapReduceUtil;
+import org.apache.hadoop.hbase.metrics.Gauge;
+import org.apache.hadoop.hbase.metrics.impl.MetricRegistriesImpl;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.lib.output.NullOutputFormat;
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import org.apache.htrace.SpanReceiver;
+import org.apache.phoenix.jdbc.PhoenixConnection;
+import org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil;
+import org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil.SchemaType;
+import org.apache.phoenix.mapreduce.util.PhoenixMapReduceUtil;
+import org.apache.phoenix.query.QueryServices;
+import org.apache.tephra.TransactionNotInProgressException;
+import org.apache.tephra.TransactionSystemClient;
+import org.apache.tephra.hbase.coprocessor.TransactionProcessor;
+import org.apache.thrift.transport.TTransportException;
+import org.apache.twill.common.Cancellable;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.discovery.ZKDiscoveryService;
+import org.apache.twill.zookeeper.ZKClient;
+import org.joda.time.Chronology;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tool to collect table level statistics on HBase snapshot
+ */
+public class UpdateStatisticsTool extends Configured implements Tool {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UpdateStatisticsTool.class);
+
+    private static final Option TABLE_NAME_OPTION = new Option("t", "table", true,
+            "Phoenix Table Name");
+    private static final Option SNAPSHOT_NAME_OPTION = new Option("s", "snapshot", true,
+            "HBase Snapshot Name");
+    private static final Option RESTORE_DIR_OPTION = new Option("d", "restore-dir", true,
```

Comment:
Do we provide default restore directory or root directory?

Reply:
I am not sure if there is an easy way of doing that. Snapshots cannot be restored to `hbase.rootDir`, which is the only configuration parameter I know for getting HDFS path. I will dig up a bit to see if default option is possible here since that will ease its operationalization.

Reply:
Regarding restore directory, is it a full path or relative path to the root path? Can it be a relative path to ease its operationalization.

Reply:
So we cannot use any path relative to `hbase.rootDir` according to the code. However after digging in, I figured out that we can use `fs.defaultFS` config to find the `hdfs.rootDir` and append `/tmp` to it, thus ensuring that we restore it in that directory always. This will help ease operationalization. Does this seem fine?

Relevant code from `RestoreSnapshotHelper` class
```
        } else if (restoreDir.toUri().getPath().startsWith(rootDir.toUri().getPath())) {
            throw new IllegalArgumentException("Restore directory cannot be a sub directory of HBase root directory. RootDir: " + rootDir + ", restoreDir: " + restoreDir);
```

Reply:
I updated the patch to reflect this new change.